### ISSUE TITLE
Hacky hour 8th October

### ISSUE THREAD

**Issue by murraycadzow** [smilefreak/studyGroup#15]
The first hacky hour where we bring along and work on our own code in a relaxed environment with the opportunity to discuss and help each other

**Comment by methylnick** [comment]
I have an interesting problem that maybe you all have come across before, and I am just a newbie at all this stuff.

SeqAnswers forum thread fyi. http://seqanswers.com/forums/newreply.php?do=postreply&t=62881

The problem is reverse engineering a fastq file from a BAM (which looks to be paired end). 

I also need to randomise the fastq file too (for every four lines of the fastq, randomise the order of these four lines in quite a big file), but! With the paired end element maintained. 

I may have an answer by that time, we shall see.

N

**Comment by smilefreak** [comment]
Ahh you fired a memory :).

Basically, I had to do this because I wanted to remap some data I was given and was worried about biasing the aligner.

http://gatkforums.broadinstitute.org/discussion/2908/howto-revert-a-bam-file-to-fastq-format

This post has the steps I followed.

**Comment by methylnick** [comment]
Thanks, my problem is a little more complicated than that, :-)

**Comment by smilefreak** [comment]
It fits exactly with your problem description, maybe you can give more explanation

Turn Fastq to BAM, but do so in a way that randomise the order of these fastq files while maintaining the paired-end relationships in the files.

**Comment by methylnick** [comment]
my bad for not explaining properly (ps hope you enjoyed the talk). BAM to fastQ, I have another fastq file from the same individual which I want to mush together, while preserving the paired end relationship, but also to randomize.

**Comment by smilefreak** [comment]
Talk was awesome, I really like the way he integrated the computer science aspects seamlessly  into the discussion about primer design. 

Ah so you have some bams that need to got into fastq? and another fastq files that you want to mush together to make one big randomised file, but making sure that pairs are randomised together?

**Comment by methylnick** [comment]
affirmative :-)

**Comment by murraycadzow** [comment]
Could you not convert your fastq into a bam then merge with your bam then go through the steps in James post?

**Comment by murraycadzow** [comment]
FYI picardtools is probably useful for fastq to sam/bam (http://broadinstitute.github.io/picard/command-line-overview.html#FastqToSam)

**Comment by smilefreak** [comment]
Anyways, I think the next port of call is an in-person discussion. Regardless of whether you solve this Nick. please bring it along to HackyHour!! Definately the kind of problem worth chatting about.

**Comment by methylnick** [comment]
yes will do James. I am sure there would be people interested in this problem.

**Comment by methylnick** [comment]
Thanks Murry for that suggestion, didn't think of it that way, there are some slight header formatting differences that also need to be resolved, I don't think I'll overcome the problem by hacky hour, happy to bring it for discussion with you all next hacky hour, it's already pencilled into my diary.

**Comment by smilefreak** [comment]
Just a reminder for HackyHour tomorrow, 3:30 in the Billiard room of the Staff Club. Let's make it a productive and fun session.

**Comment by TomKellyGenetics** [comment]
As mentioned elsewhere, I've got a friend with problems working on chromosomal locus data. They're trying to analyse co-location of CRISPR sites, sound like something we can help with?

**Comment by murraycadzow** [comment]
Can't make any promises without knowing the data but should at least have enough experience in the room to at least suggest a starting point

**Comment by methylnick** [comment]
ditto, I have pencilled in an hour but would think it would be fun to see the problem.

**Comment by methylnick** [comment]
Almost there:

` awk  '{ gsub("/1", "", $1) ; gsub("/2", "", $1) ; print }' N.sam`

but:

`awk  '{ gsub("/1", "", $1) ; gsub("/2", "", $1) ; print }' N.sam | samtools view -b - > test.bam`

returns:

`[W::sam_read1] parse error at line 1
[main_samview] truncated file.`

it seems to have removed the tabs and replaced them with whitespaces, no longer a SAM

More to come.

**Comment by TomKellyGenetics** [comment]
This issue was moved to TomKellyGenetics/studyGroup#5
